examples/rln-js/index.js

```diff
@@ -0,0 +1,500 @@
+  let membershipId;
+  let membershipKey;
+
+  ui.onManualImport((id, key) => {
+    membershipId = id;
+    membershipKey = key;
+
+    result.encoder = new RLNEncoder(
+      createEncoder({
+        ephemeral: false,
+        contentTopic: ContentTopic,
+      }),
+      rlnInstance,
+      membershipId,
+      membershipKey
+    );
+
+    ui.setMembershipInfo(membershipId, membershipKey);
+    ui.enableDialButton();
+  });
+
+  ui.onWalletImport(async () => {
+    const signer = provider.getSigner();
+
+    signature = await signer.signMessage(SIGNATURE_MESSAGE);
+    membershipKey = await rlnInstance.generateSeededMembershipKey(signature);
+
+    const idCommitment = ethers.utils.hexlify(membershipKey.IDCommitment);
+
+    rlnContract.members.forEach((m) => {
+      if (m.pubkey._hex === idCommitment) {
+        membershipId = m.index.toString();
+      }
+    });
+
+    if (membershipId) {
+      result.encoder = new RLNEncoder(
+        createEncoder({
+          ephemeral: false,
+          contentTopic: ContentTopic,
+        }),
+        rlnInstance,
+        membershipId,
+        membershipKey
+      );
+    }
+
+    ui.setMembershipInfo(membershipId, membershipKey);
+
+    const network = await provider.getNetwork();
+    ui.enableRegisterButtonForGoerli(network.chainId);
+  });
+
+  ui.onRegister(async () => {
+    let event;
+
+    ui.setRlnStatus("Trying to register...");
+    if (signature) {
+      event = await rlnContract.registerMember(rlnInstance, signature);
+    } else {
+      // if user imports credentials manually
+      // TODO(@weboko): add logic to cover it in RLNContract
+      const idCommitment = membershipKey.IDCommitment;
+      const reversedArray = idCommitment.slice().reverse();
+      const pubkey = ethers.utils.hexlify(reversedArray).toString();
+
+      const price = await rlnContract.MEMBERSHIP_DEPOSIT();
+
+      const signer = provider.getSigner();
+      const rlnContractWithSigner = rlnContract.connect(signer);
+
+      const txResponse = await rlnContractWithSigner.register(pubkey, {
+        value: price,
+      });
+      console.log("Transaction broadcasted:", txResponse);
+
+      const txReceipt = await txResponse.wait();
+
+      console.log("Transaction receipt", txReceipt);
+
+      event = txReceipt.events[0];
+    }
+
+    // Update membershipId
+    membershipId = event.args.index.toNumber();
+
+    console.log(
+      "Obtained index for current membership credentials",
+      membershipId
+    );
+
+    ui.setRlnStatus("Successfully registered.");
+    ui.setMembershipInfo(membershipId, membershipKey);
+    ui.enableDialButton();
+  });
+
+  return result;
+}
+
+async function initWaku(ui, rln) {
+  ui.setWakuStatus("Creating Waku node.");
+  const node = await createLightNode();
+  ui.setWakuStatus("Starting Waku node.");
+
+  await node.start();
+  ui.setWakuStatus("Waku node started.");
+
+  const verifyMessage = (message) => {
+    if (message.proofState === "verifying...") {
```

Comment:
While debugging the example I noticed that `rateLimitProof` is never received. 
After checking I do see it being [defined on the proto](https://github.com/waku-org/js-waku/blob/0a80cd431595b0c4b2cd94e1981aac62ac46a1cc/packages/proto/src/lib/message.proto#L21) but I do not see being mentioned in the [RFC](https://rfc.vac.dev/spec/14/#wire-format) or [proto references](https://github.com/vacp2p/waku/blob/main/waku/message/v1/message.proto).

@fryorcraken is it something expected? Maybe I am missing some knowledge here

Reply:
Best to ask @rymnc or @richard-ramos  :)

Reply:
Since RLN is experimental, and not enabled on the test fleet, you will not receive the RateLimitProof field. Please refer to the waku-rln-relay rfc https://rfc.vac.dev/spec/17 for the field addition to WakuMessage. 
When we transition from experimental, we will add the proto definition to the wire format repo :)

Reply:
Thanks for clarification @rymnc !